src/sentry/api/endpoints/project_artifact_bundle_file_details.py

```diff
@@ -0,0 +1,108 @@
+from sentry.models import ArtifactBundle, ArtifactBundleArchive, ProjectArtifactBundle
+
+
+class ProjectArtifactBundleFileDetailsMixin:
+    @classmethod
+    def download_file_from_artifact_bundle(
+        cls, file_path: str, archive: ArtifactBundleArchive
+    ) -> Union[Response, FileResponse]:
+        try:
+            fp, headers = archive.get_file_by_file_path(file_path)
+            file_info = archive.get_file_info(file_path)
+        except Exception:
+            # In case we fail here, we want to close the fail before returning.
+            archive.close()
+            return Response(
+                {"error": f"The file {file_path} can't be found in the artifact bundle"},
+                status=404,
+            )
+
+        response = FileResponse(
+            ClosesDependentFiles(fp, archive),
+            content_type=headers.get("content-type", "application/octet-stream"),
+        )
+        response["Content-Length"] = file_info.file_size if file_info is not None else None
+        response["Content-Disposition"] = 'attachment; filename="%s"' % posixpath.basename(
+            " ".join(file_path.split())
+        )
+
+        return response
+
+
+@region_silo_endpoint
+class ProjectArtifactBundleFileDetailsEndpoint(
+    ProjectEndpoint, ProjectArtifactBundleFileDetailsMixin
+):
+    permission_classes = (ProjectReleasePermission,)
+
+    def get(self, request: Request, project, bundle_id, file_id) -> Response:
+        """
+        Retrieve the file of an artifact bundle
+        `````````````````````````````````
+
+        Return details on an individual file within a release.  This does
+        not actually return the contents of the file, just the associated
+        metadata.
+
+        :pparam string organization_slug: the slug of the organization the
+                                          release belongs to.
+        :pparam string project_slug: the slug of the project to retrieve the
+                                     file of.
+        :pparam string bundle_id: the bundle_id of the artifact bundle that
+                                    should contain the file identified by file_id.
+        :pparam string file_id: the ID of the file to retrieve.
+        :auth: required
+        """
+        if not has_download_permission(request, project):
+            return Response(
+                {"error": "You don't have the permissions to download this file"}, status=403
+            )
+
+        try:
+            artifact_bundle = ArtifactBundle.objects.get(
+                organization_id=project.organization.id, bundle_id=bundle_id
+            )
+        except ArtifactBundle.DoesNotExist:
+            return Response(
+                {"error": f"The artifact bundle with {bundle_id} does not exist"}, status=404
+            )
+
+        try:
+            ProjectArtifactBundle.objects.get(
+                project_id=project.id, artifact_bundle=artifact_bundle
```

Comment:
```suggestion
                project_id=project.id, artifact_bundle__bundle_id= bundle_id
```
AFAIK you can use I DB query instead of two here:
```
In [3]: ProjectArtifactBundle.objects.filter(project_id=1, artifact_bundle__bundle_id=1)
Out[3]: <BaseQuerySet []>

In [4]:
```

Reply:
```
In [4]: print(qs.query)
SELECT "sentry_projectartifactbundle"."id", "sentry_projectartifactbundle"."organization_id", "sentry_projectartifactbundle"."project_id", "sentry_projectartifactbundle"."artifact_bundle_id", "sentry_projectartifactbundle"."date_added" FROM "sentry_projectartifactbundle" INNER JOIN "sentry_artifactbundle" ON ("sentry_projectartifactbundle"."artifact_bundle_id" = "sentry_artifactbundle"."id") WHERE ("sentry_artifactbundle"."bundle_id" = 00000000-0000-0000-0000-000000000001 AND "sentry_projectartifactbundle"."project_id" = 1)
```